### PR TITLE
Add a config class to provide some customizability

### PR DESCRIPTION
**Pull request by borkweb**
This PR adds a `StellarWP\DB\Config` class that is merely a string store, so it is all a pile of static shenanigans. Added docs on initialization and configuration.

Additionally, I tacked on the work to convert tric to slic.

### CODE REVIEW

**Comment by borkweb** [comment]
~Whoops, I guess I need to iterate on the `slic` stuff. Switching back to draft.~ **Done.**

File: src/DB/Config.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	/**
+	 * Sets the DatabaseQueryException class.
+	 *
+	 * @param string $class Class name of the DatabaseQueryException to use.
+	 *
+	 * @return void
+	 */
+	public static function setDatabaseQueryException( string $class ) {
+		self::$databaseQueryException = $class;
+	}
```
Comment (JasonTheAdams) [inline src/DB/Config.php]: @borkweb We either need to create a `DatabaseQueryException` interface to enforce the constructor and two methods and check for that, or else enforce that the class is a child class of `DatabaseQueryException`. The `DB` class makes assumptions about the data provided to the exception.

Comment (borkweb) [reply src/DB/Config.php]: Ah yes. Before, I avoided doing that due to the structure that it was marching toward. A situation where a project might have included `db` and `schema` separately, causing there to be two completely different `db` libraries in place - meaning 2+ different parent classes or interfaces.

Now that the approach has been restructured, checking and enforcing that makes a lot of sense!

Comment (borkweb) [reply src/DB/Config.php]: Added a conditional and `throw`.

File: src/DB/Config.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	/**
+	 * Sets the DatabaseQueryException class.
+	 *
+	 * @param string $class Class name of the DatabaseQueryException to use.
+	 *
+	 * @return void
+	 */
+	public static function setDatabaseQueryException( string $class ) {
+		if ( ! is_a( $class, DatabaseQueryException::class, true ) ) {
+			throw new \InvalidArgumentException( 'The provided DatabaseQueryException class must be or must extend ' . __NAMESPACE__ . '\Database\Exceptions\DatabaseQueryException.' );
```
Comment (JasonTheAdams) [inline src/DB/Config.php]: @borkweb Out of curiosity, why spell out the whole FQCN here and not just use `DatabaseQueryException::class` again?

Comment (borkweb) [reply src/DB/Config.php]: I like typing

Comment (borkweb) [reply src/DB/Config.php]:
```suggestion
			throw new \InvalidArgumentException( 'The provided DatabaseQueryException class must be or must extend ' . DatabaseQueryException::class . '.' );
```

**Review by JasonTheAdams — COMMENTED** [review]
@borkweb Do we want to add some tests for this config? Also, in case it's useful, I tend to separate static facades from the underlying class instance — we have a [Facade](https://github.com/impress-org/givewp/blob/develop/src/Framework/Support/Facades/Facade.php) class for this. That way the underlying class instance can be easily swapped out for testing. Otherwise, dealing with static properties in testing can be a bear — though I'm sure you already know this. 😃

**Comment by borkweb** [comment]
@JasonTheAdams - I _tried_ to make a half-step to minimize the extra class and interface, but that actually made things _more_ complex when trying to mock rather than extend...Soooooo, I just went with your suggestion and `Config` is now a proper Facade and we have a new class (`ConfigComponents`) to offload the static calls to. Added a pile of tests for all of that.

I don't like the readability of this, though.

**Comment by borkweb** [comment]
I just chatted with @lucatume and he showed me a nice way to have the super approachable class while making the statics testable. I'm going to give that a try and see if I can dial it back to testable simplicity.

---

**Update:** The `@backupStaticAttributes` annotation in PHPUnit didn't play nicely with some Codeception components, so I'm _not_ implementing it this way.

**Comment by borkweb** [comment]
Alrighty. This approach brings us back to a super-simple `Config` class without the Facade approach, for readability purposes. This gives us something that engineers that are not deeply steeped in OOP can look at and understand while also giving us a testable object. Approachable and armored code 👍 

I _did_ attempt to leverage `@backupStaticAttributes` in PHPUnit, but `Codeception\Util\Annotation::reflectedClasses` was a problem and blacklisting that static attribute didn't work as expected. So, instead, I've opted to add `Config::reset()`, which brings the attributes back to their original state. That method is being invoked within the test class `::tearDown()` method.

**Review by JasonTheAdams — APPROVED** [review]
Glad you were able to get the tests working! Glad to have the config in place!

**Pull request merged**